### DISCUSSION TITLE
Gif, lossy compression, but preserve alpha

### DISCUSSION THREAD

**Discussion by motionmixtape** [ImageMagick/ImageMagick#7112] [Help]
### ImageMagick version

7.1.1-27

### Operating system, version and so on

MacOS 14.2.1

### Description

I am compressing gifs using -coalesce -layers 'optimize' with great results. 

If I open the outputted gif in Photoshop I can see many of the layers/frames have transparency ([example screenshot](https://capture.dropbox.com/ZnCzqWVi0NlJ5XU2)). 

What I'd liek to do is also perform lossy compression on the gif and I don't believe ImageMagick supports this. 

I tried applying gifsicle's -lossy feature to the outputted gif. I also tried uploading the outputted gif to compress-or-die.com, adjusting the lossy slider then downloading the results.

In both instances, the transparent areas are all flattened back into having pixels. Defeating the purpose using -layers 'optimize'. 

I also also tried doing the lossy compression first, then running -layers 'optimize'. There was little difference in file size before and after.

Does anyone have any thoughts on a way to perform lossy compression (with any tool) while preserving ImageMagick's -layers 'optimize' work? basically compress the pixels, preserve the alpha?

**Comment by snibgo** [comment]
I haven't heard of a lossy GIF compression. IM has no facility to change compression of GIF files.

However, reducing the number of colors in an image before writing it to GIF can result in smaller GIF files, and could be considered a "lossy compression".

For example:
```
D:\temp>%IMG7%magick toes.png x.gif && dir x.gif

16/02/2024  19:46            44,252 x.gif


D:\temp>%IMG7%magick toes.png -colors 4 x.gif && dir x.gif

16/02/2024  19:46             5,712 x.gif
```
Reducing to 4 colours has greatly reduced the size of the GIF.

**Comment by fmw42** [comment]
One thought.  You could convert to some other format, perhaps TIFF and do JPG compression. Then convert back to GIF.

**Comment by motionmixtape** [comment]
I believe I solved it, I use ImageMagick and run '-coalesce -layers 'optimize'

I take those outputs and run them through gifsicle using -Okeep-empty to preserve transparent pixels and --lossy to compress the remaining opaque pixels
